### PR TITLE
Feature/swift 3

### PR DESCRIPTION
**Pull request by carobeta**
Migrate to Swift 3

### CODE REVIEW

**Comment by zetxek** [comment]
I have referenced the pod with:

` pod 'HanekeSwift', :git => 'https://github.com/beta-uy/HanekeSwift.git', :branch => 'feature/swift-3'`

I have also set "Use legacy swift language version" to "No", to make it build with swift 3.0:
image

Unfortunately when I build using XCode 8.0 beta 6, I get these errors:

image

Is this expected?

**Comment by carobeta** [comment]
This version is not updated for 6 beta yet. Between tomorrow and Tuesday I hope to make a new commit.

**Comment by carobeta** [comment]
@dcharbonnier Unfortunately this does not really seem to work on travis, but the xcodebuild command works locally. Can you confirm this?

**Comment by Maryom** [comment]
Hello,
Thanks for your efforts, when Swift 3 support will be available?

File: Haneke/Fetcher.swift [hunk 1]
```diff
@@ -17,7 +17,7 @@ public class Fetcher<T : DataConvertible> {
         self.key = key
     }
     
-    public func fetch(failure fail: ((NSError?) -> ()), success succeed: (T.Result) -> ()) {}
+    public func fetch(failure fail: ((NSError?) -> ()), success succeed: @escaping (T.Result) -> ()) {}
```
Comment (lluisgerard) [inline Haneke/Fetcher.swift]: `failure` should have `@escaping` as well. This happens on a lot of places in this PR, specially on fails.

**Comment by dcharbonnier** [comment]
There is other PR with more progress, closing

**Pull request closed without merging**